### PR TITLE
[ADD] module odoo_backup_sh

### PR DESCRIPTION
**Pull request by ufaks**

### CODE REVIEW

File: odoo_backup_sh/README.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
+===============
+ Backup Service
+===============
+
+Backup Service Summary
+
+{Then add more detailed description, technical specifications, any other information that could be interested for other developers. Don't forget that Usage instructions is a separated and has to be located in doc/index.rst file.}
```
Comment (yelizariev) [inline odoo_backup_sh/README.rst]: *add more detailed description*

File: odoo_backup_sh/README.rst [hunk 2]
```diff
@@ -0,0 +1,41 @@
+===============
+ Backup Service
```
Comment (yelizariev) [inline odoo_backup_sh/README.rst]: I think to use ``Odoo-backup.sh`` as a name (and brand)

File: odoo_backup_sh/README.rst [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+===============
```
Comment (yelizariev) [inline odoo_backup_sh/README.rst]: Wrong format. Two more ``=`` are needed

File: odoo_backup_sh/__manifest__.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+{
+    "name": """Backup Service""",
+    "summary": """Backup Service Summary""",
```
Comment (yelizariev) [inline odoo_backup_sh/__manifest__.py]: What about ``Remote backup service`` for the summary?

File: odoo_backup_sh/__manifest__.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    "category": "Extra Tools",
```
Comment (yelizariev) [inline odoo_backup_sh/__manifest__.py]: Use **Backup** for category

File: odoo_backup_sh/__manifest__.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    "external_dependencies": {"python": [], "bin": []},
+    "data": [
+        # "{FILE1}.xml",
+        # "{FILE2}.xml",
+    ],
+    "qweb": [
+        # "static/src/xml/{QWEBFILE1}.xml",
+    ],
+    "demo": [
+        # "demo/{DEMOFILE1}.xml",
```
Comment (yelizariev) [inline odoo_backup_sh/__manifest__.py]: lint error

File: odoo_backup_sh/controllers/main.py [hunk 7]
```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+
```
Comment (yelizariev) [inline odoo_backup_sh/controllers/main.py]: Add authorship line with your name to code files and license line to each file

File: odoo_backup_sh/doc/index.rst [hunk 8]
```diff
@@ -0,0 +1,34 @@
+===============
+ Backup Service
+===============
+
+Installation
+============
+{Instruction about things to do before actual installation}
```
Comment (yelizariev) [inline odoo_backup_sh/doc/index.rst]: Where is instruction?

File: odoo_backup_sh/doc/index.rst [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+* `Install <https://odoo-development.readthedocs.io/en/latest/odoo/usage/install-module.html>`__ this module in a usual way
+* {OPTIONAL }`Activate longpolling <https://odoo-development.readthedocs.io/en/latest/admin/longpolling.html>`__
```
Comment (yelizariev) [inline odoo_backup_sh/doc/index.rst]: Optional?

File: odoo_backup_sh/doc/index.rst [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+ Backup Service
+===============
+
+Installation
+============
+{Instruction about things to do before actual installation}
+
+* `Install <https://odoo-development.readthedocs.io/en/latest/odoo/usage/install-module.html>`__ this module in a usual way
+* {OPTIONAL }`Activate longpolling <https://odoo-development.readthedocs.io/en/latest/admin/longpolling.html>`__ 
+* {Additional notes if any}
```
Comment (yelizariev) [inline odoo_backup_sh/doc/index.rst]: Additional notes?

File: odoo_backup_sh/doc/index.rst [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+
+Configuration
+=============
+
+{Instruction how to configure the module before start to use it}
```
Comment (yelizariev) [inline odoo_backup_sh/doc/index.rst]: How to configure?

File: odoo_backup_sh/doc/index.rst [hunk 12]
```diff
@@ -0,0 +1,34 @@
...
+
+* `Activate Developer Mode <https://odoo-development.readthedocs.io/en/latest/odoo/usage/debug-mode.html>`__
+* Open menu ``[[ {Menu} ]] >> {Submenu} >> {Subsubmenu}``
+* Click ``[{Button Name}]``
+
+Usage
+=====
+
+{Instruction for daily usage. It should describe how to check that module works. What shall user do and what would user get.}
```
Comment (yelizariev) [inline odoo_backup_sh/doc/index.rst]: Instruction, please?

File: odoo_backup_sh/doc/index.rst [hunk 13]
```diff
@@ -0,0 +1,34 @@
...
+{Instruction for daily usage. It should describe how to check that module works. What shall user do and what would user get.} 
+
+* Open menu ``{Menu} >> {Submenu} >> {Subsubmenu}``
+* Click ``[{Button Name}]``
+* RESULT: {what user gets, how the modules changes default behaviour}
+
+Uninstallation
+==============
+
+{Optional section for uninstallation notes. Delete it if you don't have notes for uninstallation.}
```
Comment (yelizariev) [inline odoo_backup_sh/doc/index.rst]: Uninstallation notes if any or remove

File: odoo_backup_sh/controllers/main.py [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+            )
+        if place_for_backup_button != -1:
+            d['list_db'] = config['list_db']
+            d['databases'] = []
+            try:
+                d['databases'] = http.db_list()
+            except odoo.exceptions.AccessDenied:
+                monodb = http.db_monodb()
+                if monodb:
+                    d['databases'] = [monodb]
```
Comment (yelizariev) [inline odoo_backup_sh/controllers/main.py]: Try to get those variables via ``res.qcontext`` (when you need them or before first call of ``find``)

File: odoo_backup_sh/controllers/main.py [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+            user_key = ''.join(random.choice(string.hexdigits) for _ in range(30))
+            config.__setitem__('odoo_backup_user_key', user_key)
+            config.save()
+        headers = {
+            'host': 'odoo-backup.sh',
+            'Content-type': 'application/json',
+            'Accept': 'text/plain'
+        }
+        res = requests.post(
+            'http://odoo_backup_sh_service/web/backup/list', headers=headers, json={'params': {'user_key': user_key}}
```
Comment (yelizariev) [inline odoo_backup_sh/controllers/main.py]: Use global variable at the beginning of the file for hostname with protocol (``http://odoo_backup_sh_service``), please

File: odoo_backup_sh/__manifest__.py [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+{
+    "name": """Backup Service""",
+    "summary": """Backup Service Summary""",
+    "category": "Extra Tools",
+    # "live_test_url": "",
+    "images": [],
+    "version": "11.0.1.0.0",
+    "application": False,
+
+    "author": "IT-Projects LLC, Ivan Yelizariev",
```
Comment (yelizariev) [inline odoo_backup_sh/__manifest__.py]: Your name is not Ivan Yelizariev

File: odoo_backup_sh/controllers/main.py (hunk 14 above)
Comment (ufaks) [reply odoo_backup_sh/controllers/main.py]: def _render_template(self, **d) returns a rendered template as unicode string (see http://jinja.pocoo.org/docs/2.10/api/). There is no the qcontext attribute.

Comment (yelizariev) [reply odoo_backup_sh/controllers/main.py]: Ok

File: odoo_backup_sh/controllers/main.py [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+                if monodb:
+                    d['databases'] = [monodb]
+            backup_button = env.get_template("backup_button.html").render(d)
+            res = res[:place_for_backup_button] + backup_button + res[place_for_backup_button:]
+        return res
+
+
+class BackupController(http.Controller):
+
+    @http.route('/web/database/backup_list', type='http', auth="none")
```
Comment (yelizariev) [inline odoo_backup_sh/controllers/main.py]: Rename it to ``/web/database/backups``

**Pull request merged**